tests/lib/Service/OpenProjectAPIServiceTest.php

```diff
@@ -750,13 +750,23 @@ private function getOpenProjectAPIService(
 				true,
 				true
 			);
+		// for nextcloud above 31 OC_Util::getVersion() method does not exists
+		if (\class_exists('OCP\ServerVersion')) {
+			$version = $this->createMock(\OCP\ServerVersion::class);
+		} else {
+			/** @psalm-suppress UndefinedMethod getVersion() method is not in stable31 so making psalm not complain */
+			$version = $this->createMock(\OC_Util::class);
+		}
+
 		//changed from nextcloud 26
+		/** @psalm-suppress TooManyArguments from NC 31+ requires 6 parameters (including ServerVersion)*/
 		$ocClient = new Client(
 			$clientConfigMock,
 			$certificateManager,
 			$client,
 			$this->createMock(IRemoteHostValidator::class),
-			$this->createMock(LoggerInterface::class));
+			$this->createMock(LoggerInterface::class),
+			$version);
 
 		$clientService = $this->getMockBuilder(IClientService::class)->getMock();
 		$clientService->method('newClient')->willReturn($ocClient);
```

Comment:
you can mock the class like this:
```suggestion
		$clientService = $this->getMockBuilder(IClientService::class)->getMock();
		$clientService->method('newClient')->willReturn($this->createMock(Client::class));
```